# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontEntry.h
# Commit: 095b3edec3c8
# Full Hash: 095b3edec3c8ba9318c71d0acc91e10ce57818dc
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-04-29 21:53:38
# Regressor Bug: 1514869
# File Overlap Count: 1
# Description:
#   Bug 1514869 - patch 2 - Adapt platform-font-list code to work with either the existing in-process font list or cross-process shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22938
# ==============================================================================

diff -r 21ef00977ab6 -r 095b3edec3c8 gfx/thebes/gfxFontEntry.h
--- a/gfx/thebes/gfxFontEntry.h	Sat Apr 27 15:37:29 2019 +0000
+++ b/gfx/thebes/gfxFontEntry.h	Sat Apr 27 15:37:58 2019 +0000
@@ -201,14 +201,18 @@
   }
 
   inline bool HasCmapTable() {
-    if (!mCharacterMap) {
+    if (!mCharacterMap && !mShmemCharacterMap) {
       ReadCMAP();
-      NS_ASSERTION(mCharacterMap, "failed to initialize character map");
+      NS_ASSERTION(mCharacterMap || mShmemCharacterMap,
+                   "failed to initialize character map");
     }
     return mHasCmapTable;
   }
 
   inline bool HasCharacter(uint32_t ch) {
+    if (mShmemCharacterMap) {
+      return mShmemCharacterMap->test(ch);
+    }
     if (mCharacterMap && mCharacterMap->test(ch)) {
       return true;
     }
@@ -401,6 +405,9 @@
 
   RefPtr<gfxCharacterMap> mCharacterMap;
 
+  mozilla::fontlist::Face* mShmemFace = nullptr;
+  const SharedBitSet* mShmemCharacterMap = nullptr;
+
   mozilla::UniquePtr<uint8_t[]> mUVSData;
   mozilla::UniquePtr<gfxUserFontData> mUserFontData;
   mozilla::UniquePtr<gfxSVGGlyphs> mSVGGlyphs;